src/ParseFile.js

```diff
@@ -137,26 +135,14 @@ class ParseFile {
           type: specifiedType,
         };
       } else if (data && typeof data.base64 === 'string') {
-        const base64 = data.base64;
-        const commaIndex = base64.indexOf(',');
-
-        if (commaIndex !== -1) {
-          const matches = dataUriRegexp.exec(base64.slice(0, commaIndex + 1));
-          // if data URI with type and charset, there will be 4 matches.
-          this._data = base64.slice(commaIndex + 1);
-          this._source = {
-            format: 'base64',
-            base64: this._data,
-            type: matches[1],
-          };
-        } else {
-          this._data = base64;
-          this._source = {
-            format: 'base64',
-            base64: base64,
-            type: specifiedType,
-          };
-        }
+        const base64 = data.base64.split(',').at(-1);
+        const type = specifiedType || data.base64.split(';').at(0).split(':').at(1) || '';
```

Comment:
This is a less strict string parsing than the previous regex. What are the benefits of removing the original regex and could a less strict parsing create potential vulnerabilities?

Reply:
I added validation to check if the string follows the scheme of a URI or if it is base64 encoded data.